### ISSUE TITLE
Token generated with BearerTokenModule has incorrect expiry time 

### ISSUE THREAD

**Issue by gavinbell** [unosquare/embedio-extras#19]
The bearer token generated has a different expiry date to that what was requested by the service provider. 

```
await context.JsonResponseAsync(new BearerToken
                    {
                        Token = validationContext.GetToken(SecretKey),
                        TokenType = "bearer",
                        ExpirationDate = authorizationServerProvider.GetExpirationDate(),
                        Username = validationContext.IdentityName,
                    });
```

where the line

```
   validationContext.GetToken(SecretKey)
```

creates a token with a default expiry date. 

I did the following to fix the issue

```
                if (validationContext.IsValidated)
                {
                    var expiryDate = DateTime.SpecifyKind(DateTime.FromBinary(authorizationServerProvider.GetExpirationDate()), DateTimeKind.Utc);

                    await context.JsonResponseAsync(new BearerToken
                    {
                        Token = validationContext.GetToken(SecretKey, expiryDate),
                        TokenType = "bearer",
                        ExpirationDate = authorizationServerProvider.GetExpirationDate(),
                        Username = validationContext.IdentityName,
                    });
                }
...

        public string GetToken(SymmetricSecurityKey secretKey, DateTime? expires = null)
        {
            var tokenHandler = new JwtSecurityTokenHandler();

            var plainToken = tokenHandler.CreateToken(new SecurityTokenDescriptor
            {
                Subject = Identity,
                Issuer = "Embedio Bearer Token",
                Expires = expires,
                SigningCredentials = new SigningCredentials(secretKey,
                    SecurityAlgorithms.HmacSha256Signature),
            });

            return tokenHandler.WriteToken(plainToken);
        }

```

I hope this helps.

**Comment by geoperez** [comment]
Cool, thanks. Let me fix it and release a new nuget.

**Comment by geoperez** [comment]
Published new nuget version 2.2.3.
